### ISSUE TITLE
Add serial filter for the token/challenges/ endpoint

### ISSUE THREAD

**Issue by tamaro-skaljic** [privacyidea/privacyidea#1701]
Clients which send a request to the token/challenges/ endpoint get a bunch of redundant informations. Cornelius thinks they will kill the response time 'cause they send this request constantly (for example each second) when 5000 users try to authenticate against privacyidea.

Here is a result from the token/challenges/ endpoint (which only contain eight authentication requests, each with two tokens...)
We must only retrieve the (e. g. push) token(s) which are related to the user instead of all authentication request tokens from each user.

```HTTP/1.1 200 OK
Cache-Control: no-cache
Connection: Keep-Alive
Content-Length: 4992
Content-Type: application/json
Date: Fri, 21 Jun 2019 10:27:22 GMT
Keep-Alive: timeout=5, max=100
Server: Apache/2.4.29 (Ubuntu)

{
    "id": 1,
    "jsonrpc": "2.0",
    "result": {
        "status": true,
        "value": {
            "challenges": [
                {
                    "challenge": "S7Z2EDLMNNR5H67Y3NTCZ2VPLQ66Q4WJ",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:28:56 GMT",
                    "id": 382,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "PIPU000006FC",
                    "timestamp": "Fri, 21 Jun 2019 10:26:56 GMT",
                    "transaction_id": "03316609898223068784"
                },
                {
                    "challenge": "",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:28:56 GMT",
                    "id": 383,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "OATH0001B59E",
                    "timestamp": "Fri, 21 Jun 2019 10:26:56 GMT",
                    "transaction_id": "03316609898223068784"
                },
                {
                    "challenge": "",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:07 GMT",
                    "id": 384,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "OATH00010A46",
                    "timestamp": "Fri, 21 Jun 2019 10:27:07 GMT",
                    "transaction_id": "03740756657316379538"
                },
                {
                    "challenge": "XVSKNBR22DXT4LFSJUG5CW5X6KC4YSSU",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:07 GMT",
                    "id": 385,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "PIPU0001B5A7",
                    "timestamp": "Fri, 21 Jun 2019 10:27:07 GMT",
                    "transaction_id": "03740756657316379538"
                },
                {
                    "challenge": "Q6TUHPHMIABAPC5WXGVXR7O3E4IBE4S3",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:10 GMT",
                    "id": 386,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "PIPU000006FC",
                    "timestamp": "Fri, 21 Jun 2019 10:27:10 GMT",
                    "transaction_id": "08696718755656752778"
                },
                {
                    "challenge": "",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:10 GMT",
                    "id": 387,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "OATH0001B59E",
                    "timestamp": "Fri, 21 Jun 2019 10:27:10 GMT",
                    "transaction_id": "08696718755656752778"
                },
                {
                    "challenge": "",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:12 GMT",
                    "id": 388,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "OATH00010A46",
                    "timestamp": "Fri, 21 Jun 2019 10:27:12 GMT",
                    "transaction_id": "08219047304346827191"
                },
                {
                    "challenge": "5VZP7EKJZT7TUKU7P3MULOJV7PJ6CMIJ",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:12 GMT",
                    "id": 389,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "PIPU0001B5A7",
                    "timestamp": "Fri, 21 Jun 2019 10:27:12 GMT",
                    "transaction_id": "08219047304346827191"
                },
                {
                    "challenge": "2VCRUHBN4F7WFENYDRMVAJHJ4PM4JPN4",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:13 GMT",
                    "id": 390,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "PIPU000006FC",
                    "timestamp": "Fri, 21 Jun 2019 10:27:13 GMT",
                    "transaction_id": "06452748753908708243"
                },
                {
                    "challenge": "",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:13 GMT",
                    "id": 391,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "OATH0001B59E",
                    "timestamp": "Fri, 21 Jun 2019 10:27:13 GMT",
                    "transaction_id": "06452748753908708243"
                },
                {
                    "challenge": "",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:15 GMT",
                    "id": 392,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "OATH00010A46",
                    "timestamp": "Fri, 21 Jun 2019 10:27:15 GMT",
                    "transaction_id": "18323780520679905896"
                },
                {
                    "challenge": "DGR4RUJ4LTQLQYIE5YVRSFUU6V2F2JZG",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:15 GMT",
                    "id": 393,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "PIPU0001B5A7",
                    "timestamp": "Fri, 21 Jun 2019 10:27:15 GMT",
                    "transaction_id": "18323780520679905896"
                },
                {
                    "challenge": "CCS3TJKMN5XOF657WOEK5BVEJZOPABGW",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:16 GMT",
                    "id": 394,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "PIPU000006FC",
                    "timestamp": "Fri, 21 Jun 2019 10:27:16 GMT",
                    "transaction_id": "01141929036262014936"
                },
                {
                    "challenge": "",
                    "data": "",
                    "expiration": "Fri, 21 Jun 2019 10:29:17 GMT",
                    "id": 395,
                    "otp_received": false,
                    "otp_valid": false,
                    "received_count": 0,
                    "serial": "OATH0001B59E",
                    "timestamp": "Fri, 21 Jun 2019 10:27:17 GMT",
                    "transaction_id": "01141929036262014936"
                },
            ],
            "count": 16,
            "current": 1,
            "next": 2,
            "prev": null
        }
    },
    "signature": "rsa_sha256_pss:censored",
    "time": 1561112842.957909,
    "version": "privacyIDEA 3.0.1",
    "versionnumber": "3.0.1"
}


```

**Comment by cornelinux** [comment]
Hi @tamaro-skaljic thanks for the issue.
We already fixed this in #1689 ;-)

You can pass the serial number like

    GET /token/challenges/<serial>

or in most cases the application (like owncloud or keycloak) will have the transaction_id, this can also be passed as parameter

    GET /token/challenges/?transaction_id=<transid>

So I am closing this issue as a duplicate.

**Comment by tamaro-skaljic** [comment]
> Hi @tamaro-skaljic thanks for the issue.
> We already fixed this in #1689 ;-)
> 
> You can pass the serial number like
> 
> ```
> GET /token/challenges/<serial>
> ```
> 
> or in most cases the application (like owncloud or keycloak) will have the transaction_id, this can also be passed as parameter
> 
> ```
> GET /token/challenges/?transaction_id=<transid>
> ```
> 
> So I am closing this issue as a duplicate.

Hey cool! Thanks!
